pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,75 @@
+{ godot-mono
+, lib
+, nuget-to-nix
+, path
+, stdenv
+}:
+
+# This script generates deps.nix in your current directory. To use it, simply do:
```

Comment:
This extra indirection just needlessly complicates things. We can just write a bash script that uses a nix-shell as the shebang. This would also make debugging a walk in the park.

Reply:
I forget why I was having the "runner" script generated as an output of make-deps.nix rather than directly dropping it into a shell script next to the derivation ... However, looking back at it, I'm not sure how I'd hard-code these two substitutions being done:
```sh
#!${stdenv.shell}
nix-shell ${path} -A godot-mono.make-deps --run 'eval "$makeDeps"'
```
If it's generated as an output, it can know the path to the copy of nixpkgs that generated it, which gets put in place of ${path}.  It can also use that nixpkgs's copy of the stdenv.shell to run it.  If I just moved it to a make-deps.sh or something containing that script, I don't know what I'd give it as the shell or nixpkgs path.

Reply:
Actually, I think I understand what to do here.

Reply:
I don't think I can use a nix-shell shebang.  All examples I can find of this use the `-p` flag to pull in explicit dependencies, for instance:
```sh
#!/usr/bin/env nix-shell
#!nix-shell -I nixpkgs=../../../../../ -i bash -p nix wget prefetch-yarn-deps nix-prefetch-github
```
However, for the godot mono make-deps script, it needs to be able to explicitly run a few phases (unpack, patch, and configure) of the godot-mono derivation, followed by some additional steps.  To get these phases, I need to use the `-A` flag of nix-shell instead, to pull in the build environment of a derivation.  Unfortunately, it doesn't look like it's possible to make a nix-shell shebang using -A.  This (and some other variations I tried) doesn't work:
```sh
#!/usr/bin/env nix-shell
#!nix-shell ../../../../../ -A godot-mono.make-deps
set -e
outdir="$(pwd)"
wrkdir="$(mktemp -d)"
pushd "$wrkdir" > /dev/null
  unpackPhase
  cd source
  patchPhase
  configurePhase

  # Without RestorePackagesPath set, it restores packages to a temp directory. Specifying
  # a path ensures we have a place to run nuget-to-nix.
  nugetRestore() { dotnet msbuild -t:Restore -p:RestorePackagesPath=nugetPackages $1; }

  nugetRestore modules/mono/glue/GodotSharp/GodotSharp.sln
  nugetRestore modules/mono/editor/GodotTools/GodotTools.sln

  nuget-to-nix nugetPackages > "$outdir"/deps.nix
popd > /dev/null
rm -rf "$wrkdir"
```
So I think the best I can do is keep the bulk of the logic inside a makeDeps phase within make-deps.nix, but then just have a make-deps.sh containing:
```sh
#!/usr/bin/env bash
nix-shell ../../../../../ -A godot-mono.make-deps --run 'eval "$makeDeps"'
```